grails-app/services/com/unifina/service/BalanceService.groovy

```diff
@@ -35,24 +40,32 @@ class BalanceService {
 	}
 }
 
-interface Web3Balance {
-	BigInteger checkBalance(String address) throws InterruptedException, ExecutionException, MessageDecodingException
+abstract class Web3Balance {
+	abstract BigInteger getDatacoinBalance(String address) throws InterruptedException, ExecutionException, MessageDecodingException
+
+	Map<String, BigInteger> getDatacoinBalances(SecUser user) throws InterruptedException, ExecutionException, MessageDecodingException  {
+		def keys =
```

Comment:
Should be done in 1 query instead of 2

Reply:
What does that look like in groovy? **IntegrationKey.findAllByUserAndService(user, IntegrationKey.Service.ETHEREUM | IntegrationKey.Service.ETHEREUM_ID )** ?